File: 프로그래머스/lv3/17676. ［1차］ 추석 트래픽/［1차］ 추석 트래픽.java
import java.util.*;

class Solution {
    int convertMs(String time) {
        String[] temp = time.split(":");
        return (int)((Integer.parseInt(temp[0])*60*60+Integer.parseInt(temp[1])*60+Double.parseDouble(temp[2]))*1000);
    }
    
    public int solution(String[] lines) {
        int answer = 0;
        List<Request> list = new ArrayList<>();
        int start = 0, end = 0;
        
        for (int i=0;i<lines.length;i++) {
            String[] temp;
            temp = lines[i].split(" ");
            end = convertMs(temp[1]);
            
            if (end == 0) {
                start = 0;
            } else {
                temp[2] = temp[2].substring(0, temp[2].length()-1);
                start = end - (int)((Double.parseDouble(temp[2]) * 1000)) + 1;
            }
            list.add(new Request(start, end));
        }
        
        int temp;
        start = convertMs(lines[0].split(" ")[1]);
        for (int i=0;i<list.size();i++) {
            temp = 1;
            for (int j=i+1;j<list.size();j++){
                if (list.get(j).start < list.get(i).end+1000) {
                    temp++;
                }
            }
            if (answer<temp){
                answer = temp;
            }
        }
        return answer;
    }
    
    class Request {
        int start;
        int end;
        
        Request(int start, int end) {
            this.start = start;
            this.end = end;
        }
    }
}

File: 프로그래머스/3/17676. ［1차］ 추석 트래픽/［1차］ 추석 트래픽.py
def solution(lines):
    logs = []
    for line in lines:
        _, time, term = line.split()
        end = int(convertToMili(time))
        start = int(end - float(term[:-1]) * 1000 + 1)
        logs.append((start, end))

    answer = 1
    for i in range(len(logs)-1):
        temp = 1
        end = logs[i][1]
        for j in range(i+1, len(logs)):
            start = logs[j][0]
            if start < end + 1000:
                temp += 1
        answer = max(answer, temp)
    return answer

def convertToMili(time):
    h, m, s = map(float, time.split(":"))
    return (h * 60 * 60 + m * 60 + s) * 1000